zaps/0002-encrypt-push-notifications.md

```diff
@@ -0,0 +1,182 @@
+- ZAP PR: [zulip/zulip-architecture#2](https://github.com/zulip/zulip-architecture/pull/2)
+
+# Summary
+
+Proposal to add end-to-end encryption for mobile push notifications.
+
+# Motivation
+
+Currently, the Zulip Push Notification Forwarding Service receives
+(but does not store or log) the contents of individual mobile push notifications.
+
+Adding end-to-end encryption for push notifications will ensure that the
+actual message content does not need to be readable by the push notification
+bouncer (or Google/Apple's servers, for that matter).
+
+# Detailed design
+
+## High-Level Plan
```

Comment:
If we think this subsection is not useful, we can remove it. I just added it as a brief overview for someone new reading it -- instead of directly jumping into data model.